pkgs/by-name/pl/plugdata/package.nix

```diff
@@ -63,71 +67,95 @@ stdenv.mkDerivation (finalAttrs: {
+      type = "Application";
+      categories = [
+        "AudioVideo"
+        "Music"
+      ];
+    })
   ];
-  # Standard fix for JUCE programs: https://github.com/NixOS/nixpkgs/blob/5014727e62ae7b22fb1afc61d789ca6ad9170435/pkgs/applications/audio/bespokesynth/default.nix#L137
-  env.NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isLinux "-rpath ${
-    lib.makeLibraryPath ([
-      xorg.libX11
-      xorg.libXrandr
-      xorg.libXinerama
-      xorg.libXext
-      xorg.libXcursor
-      xorg.libXrender
-    ])
-  }";
-  dontPatchELF = true; # needed or nix will try to optimize the binary by removing "useless" rpath
 
-  postPatch = ''
-    # Don't build LV2 plugin (it hangs), and don't automatically install
-    sed -i 's/ LV2 / /g' CMakeLists.txt
+  NIX_LDFLAGS = (
+    toString [
+      "-lX11"
+      "-lXext"
+      "-lXcomposite"
+      "-lXcursor"
+      "-lXinerama"
+      "-lXrandr"
+      "-lXtst"
+      "-lXdmcp"
+    ]
+  );
+
+  preBuild = ''
+    # fix LV2 build
+    HOME=$(mktemp -d)
   '';
 
   installPhase = ''
     runHook preInstall
 
     cd .. # build artifacts are placed inside the source directory for some reason
-    mkdir -p $out/{bin,lib/{clap,vst3}}
+    mkdir -p $out/bin $out/lib/clap $out/lib/lv2 $out/lib/vst3
     cp    Plugins/Standalone/plugdata      $out/bin
     cp -r Plugins/CLAP/plugdata{,-fx}.clap $out/lib/clap
     cp -r Plugins/VST3/plugdata{,-fx}.vst3 $out/lib/vst3
+    cp -r Plugins/LV2/plugdata{,-fx}.lv2   $out/lib/lv2
 
-    icon_name="plugdata_logo.png"
-    icon_path="Resources/Icons/$icon_name"
-
-    install -m644 -D "${desktopItem}"/share/applications/* -t $out/share/applications
-    for size in 16 24 32 48 64 128 256 512; do
-      mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
-      convert -resize "$size"x"$size" "$icon_path" $out/share/icons/hicolor/"$size"x"$size"/apps/"$icon_name"
-    done
+    install -Dm444 $src/Resources/Icons/plugdata_logo_linux.png $out/share/pixmaps/plugdata_logo.png
 
     runHook postInstall
   '';
 
   postInstall = ''
     # Ensure zenity is available, or it won't be able to open new files.
-    # These X11 libs get dlopen'd, they cause visual bugs when unavailable.
     wrapProgram $out/bin/plugdata \
       --prefix PATH : '${
         lib.makeBinPath [
           zenity
         ]
-      }' \
-      --prefix LD_LIBRARY_PATH : '${
-        lib.makeLibraryPath [
-          xorg.libXrandr
-          xorg.libXinerama
-          xorg.libXcursor
-          xorg.libXrender
-        ]
       }'
   '';
 
-  meta = with lib; {
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
     description = "Plugin wrapper around Pure Data to allow patching in a wide selection of DAWs";
     mainProgram = "plugdata";
     homepage = "https://plugdata.org/";
-    license = licenses.gpl3;
-    platforms = platforms.linux;
-    maintainers = with maintainers; [ PowerUser64 ];
+    license = lib.licenses.gpl3;
```

Comment:
Upstream does not specify GPL3-only vs. GPL3-plus. Seems okay to leave as is for now.